src/Moryx.Products.Management/Modification/ProductConverter.cs

```diff
@@ -74,12 +74,23 @@ public ProductModel ConvertProduct(IProductType productType, bool flat)
             return converted;
         }
 
-        private static ProductFile[] ConvertFiles(IProductType productType, IEnumerable<PropertyInfo> properties)
+        private static ProductFileModel[] ConvertFiles(IProductType productType, IEnumerable<PropertyInfo> properties)
         {
-            var files = (from property in properties
-                         where property.PropertyType == typeof(ProductFile)
-                         select (ProductFile)property.GetValue(productType)).ToArray();
-            return files;
+            var productFileProperties = properties.Where(p => p.PropertyType == typeof(ProductFile)).ToArray();
+            var fileModels = new ProductFileModel[productFileProperties.Length];
+            for (int i = 0; i < fileModels.Length; i++)
+            {
+                var value = (ProductFile)productFileProperties[i].GetValue(productType);
+                fileModels[i] = new ProductFileModel()
+                {
+                    PropertyName = productFileProperties[i].Name,
+                    FileName = value?.Name,
```

Comment:
You can replace the type, but you can not rename properties without a breaking API change.

Reply:
Even though the ProductConverter is an internal type? 🤔

Reply:
Resolved. As the ProductModel is serialized and provided via the web service, removing the Files Property would be a breaking change.